### PR TITLE
Namespaces spring configuration to match package names

### LINKED ISSUE #239: Make AutoConfiguration tests
AutoConfiguration packages include property classes that map to configuration. Currently, the only way for us to find any bugs in the mapping is to manually set these and watch it work or fail.

This needs to be hardened before we go to version 1.0 as it is too easy to create config bugs.

### PR DESCRIPTION
**Pull request by adriancole**
This is a second go at namespacing spring configuration. This time, it
includes tests. I've also manually built docker to make sure we didn't
break it.

See #235 (for last attempt)
Fixes #239

### CODE REVIEW

**Comment by adriancole** [comment]
cc @shakuzen @marcingrzejszczak

File: zipkin-autoconfigure/collector-scribe/src/test/java/zipkin/autoconfigure/collector/scribe/ZipkinScribeCollectorAutoConfigurationTests.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    addEnvironment(context, "zipkin.collector.scribe.port", "9999");
+    context.register(PropertyPlaceholderAutoConfiguration.class,
+        ZipkinScribeCollectorAutoConfiguration.class, InMemoryConfiguration.class);
+    context.refresh();
+
+    assertThat(context.getBean(ZipkinScribeCollectorProperties.class).getPort())
+        .isEqualTo(9999);
+  }
+
+  static void addEnvironment(AnnotationConfigApplicationContext context, String... pairs) {
```
Comment (shakuzen) [inline zipkin-autoconfigure/collector-scribe/src/test/java/zipkin/autoconfigure/collector/scribe/ZipkinScribeCollectorAutoConfigurationTests.java]: Why not use the `EnvironmentTestUtils` to do this? Like [this](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/PropertyPlaceholderAutoConfigurationTests.java#L52)

Comment (adriancole) [inline zipkin-autoconfigure/collector-scribe/src/test/java/zipkin/autoconfigure/collector/scribe/ZipkinScribeCollectorAutoConfigurationTests.java]: If that's in some published jar, works for me.

Comment (shakuzen) [inline zipkin-autoconfigure/collector-scribe/src/test/java/zipkin/autoconfigure/collector/scribe/ZipkinScribeCollectorAutoConfigurationTests.java]: Looks like it is included in the `spring-boot` jar, so we already have it available in this code without any dependency changes.

Comment (adriancole) [inline zipkin-autoconfigure/collector-scribe/src/test/java/zipkin/autoconfigure/collector/scribe/ZipkinScribeCollectorAutoConfigurationTests.java]: Oh nice. Ill switch to this then

Comment (adriancole) [inline zipkin-autoconfigure/collector-scribe/src/test/java/zipkin/autoconfigure/collector/scribe/ZipkinScribeCollectorAutoConfigurationTests.java]: done! Thanks for the recommendation

**Pull request merged**